### ISSUE TITLE
trouble with let

### ISSUE THREAD

**Issue by ecavallo** [jonsterling/cooltt#58]
```
def simple : {
  (A : univ) (a : A) -> A
} = {
  \A a =>
  let b : A = a in
  b
}

normalize simple
```
raises ```Internal error (Failed to normalize): do_sub_out```.

I failed to track down the reason. (The same thing happens on the pathcoe or fhcom branches btw.)

**Comment by jonsterling** [comment]
Looks pretty subtle. I'll fix it.

**Comment by jonsterling** [comment]
I found the problem, I'll push my fix once #57 is merged.
